app/scripts/metamask-controller.test.js

```diff
@@ -836,63 +836,102 @@ describe('MetaMaskController', () => {
         ).toStrictEqual([]);
       });
 
-      it('returns the eth accounts from the CAIP-25 permission', async () => {
-        jest
-          .spyOn(metamaskController.permissionController, 'getCaveat')
-          .mockReturnValue({
-            value: {
-              requiredScopes: {},
-              optionalScopes: {
-                'eip155:1': {
-                  accounts: ['eip155:1:0xdead', 'eip155:1:0xbeef'],
+      describe('the wallet is locked', () => {
+        beforeEach(() => {
+          jest.spyOn(metamaskController, 'isUnlocked').mockReturnValue(false);
+        });
+
+        it('returns empty array if there is a CAIP-25 permission for the origin and ignoreLock is false', async () => {
+          jest
+            .spyOn(metamaskController.permissionController, 'getCaveat')
+            .mockReturnValue({
+              value: {
+                requiredScopes: {},
+                optionalScopes: {
+                  'eip155:1': {
+                    accounts: ['eip155:1:0xdead', 'eip155:1:0xbeef'],
+                  },
                 },
               },
-            },
-          });
-
-        expect(
-          metamaskController.getPermittedAccounts('test.com'),
-        ).toStrictEqual(['0xdead', '0xbeef']);
-      });
-    });
+            });
 
-    describe('#getPermittedAccountsSorted', () => {
-      it('gets the permitted accounts for the origin', async () => {
-        jest
-          .spyOn(metamaskController, 'getPermittedAccounts')
-          .mockReturnValue([]);
+          expect(
+            metamaskController.getPermittedAccounts('test.com', false),
+          ).toStrictEqual([]);
+        });
 
-        await metamaskController.getPermittedAccountsSorted('test.com');
+        it('returns accounts if there is a CAIP-25 permission for the origin and ignoreLock is true', async () => {
+          jest
+            .spyOn(metamaskController.permissionController, 'getCaveat')
+            .mockReturnValue({
+              value: {
+                requiredScopes: {},
+                optionalScopes: {
+                  'eip155:1': {
+                    accounts: ['eip155:1:0xdead', 'eip155:1:0xbeef'],
+                  },
+                },
+              },
+            });
+          jest
+            .spyOn(metamaskController, 'sortAccountsByLastSelected')
+            .mockResolvedValue(['not_empty']);
 
-        expect(metamaskController.getPermittedAccounts).toHaveBeenCalledWith(
-          'test.com',
-        );
+          expect(
+            metamaskController.getPermittedAccounts('test.com', true),
+          ).not.toStrictEqual([]);
+        });
```

Comment:
🤔 are we able to make a positive assertion about what this returns?

Reply:
overlaps with the happy case below, but yeah i suppose that would be better

Reply:
https://github.com/MetaMask/metamask-extension/pull/28855/commits/3c8b24b34b3728b654f5d53809da768a723317ce#diff-0aea0fd14a27750ddf078dd4ef363f4d10b91285954ad5e1488edcad1d5bd306R882